### DISCUSSION TITLE
​Import tags from Obsidian Markdown files

### DISCUSSION THREAD

**Discussion by lucaszischka** [toeverything/AFFiNE#6176] [Ideas]
### Description

First of all I hope that I haven't overlooked an issue that is already requesting this; also please transfer it to blocksuite if I opened it at the wrong place.

---

Obsidian stores its tags at the top of the file like this <sup>[1](https://help.obsidian.md/Editing+and+formatting/Tags)</sup>:
```
---
tags:
  - tag1
  - tag2
aliases:
  - ...
---

# Content of markdown file
```

To my knowledge the singular (tag, alias) can also occur.

As far as I can tell you currently don't support aliases, but I wanted to include them in the example so that you can account for them.

---

When importing a markdown file it would be great if they get converted and then removed from the top of the file automatically.
Of course you could ask the user if they would like to automatically convert the tags, in case you think that this behaviour is unwanted for some users by default.

---

Anyway thanks for your work on this amazing project ❤️!

### Use case

Users who switch from obsidian would otherwise need to create and assign their tags for every file individually. It could therefore make it easier for obsidian users to switch.

### Anything else?

I think other markdown-only editors will have to include their tags somewhere as well, but I unfortunately don't have information on this.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by kuyper** [comment]
In addition, in obsidian, users can also define tags in the markdown instead of using front matter. For my use case, this is the more common way.
